cmake/platform.cmake

```diff
@@ -137,7 +137,13 @@ elseif(UNIX OR MINGW)
             append(CMAKE_CCXX_SANITIZER_FLAGS "-g -fno-omit-frame-pointer")
         endif()
     elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
-        set(DEF_ARCH_OPT_FLAGS "-msse4.1")
+        if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(aarch64.*|AARCH64.*|arm64.*|ARM64.*)")
```

Comment:
maybe introduce `TARGET_ARCH=X86_64|AARCH64` to avoid duplication of this magic expression?

Reply:
To be a little bit more specific, let's put the following around `/CMakeLists.txt:68` (where we check for the pointer size):
``` cmake
if (CMAKE_SYSTEM_PROCESSOR MATCHES "^(aarch64.*|AARCH64.*|arm64.*|ARM64.*)")
    set(TARGET_ARCH AARCH64)
else()
    set(TARGET_ARCH X86_64)
endif()
```

Reply:
Thanks for the suggestion - I'm just getting ready to push push some changes to do this, and address the other comments, should be with you soon.

Reply:
Done, just after the CMAKE_SIZEOF_VOID_P test.